controller/engine_controller.go

```diff
@@ -1336,13 +1329,15 @@ func (ec *EngineController) removeUnknownReplica(e *longhorn.Engine) error {
 		return nil
 	}
 
-	engineClientProxy, err := ec.getEngineClientProxy(e, e.Status.CurrentImage)
-	if err != nil {
-		return err
-	}
-
 	for _, url := range unknownReplicaURLs {
 		go func(url string) {
+			engineClientProxy, err := ec.getEngineClientProxy(e, e.Status.CurrentImage)
+			if err != nil {
+				ec.logger.WithError(err).Warnf("Failed to remove unknown replica %v from engine", url)
```

Comment:
Should we use `Errorf()` here?

Reply:
This is a client retrieval error not related the actual `ReplicaRemove`. I feel this should be in warning because it might resolve itself on the next queue. WDYT?

Reply:
I see. Thank you.

Reply:
Error this outside of goroutine so it gets enqueued again.